plugins/builder/osbuild.py

```diff
@@ -208,11 +262,39 @@ def parse_certs(string):
 
         return certs
 
+    def oauth_init(self, client_id: str, secret: str, token_url: str):
+        oauth = OAuth2(client_id, secret, token_url)
+        self.http.auth = oauth
+
+    def oauth_check(self) -> bool:
+        auth = self.http.auth
+        if auth and auth.token_expired:
+            auth.fetch_token(self.http)
+            return True
+
+        return False
+
+    def request(self, method: str, url: str, js: Optional[Dict] = None):
+
+        self.oauth_check()
+        res = self.http.request(method, url, json=js)
+
+        if res.status_code != 401 and self.oauth_check():
+            res = self.http.request(method, url, json=js)
```

Comment:
Hm wait now it won't retry on 401s the first time? The lazy refresh is definitely a better approach but on 401s you'd wanna retry after redoing the oauth_check no? And return early if 2xx.

Reply:
Doh. Inverted logic since I did it the other way before. Will fix!

Reply:
Added a test for that case and fixed the logic.